pkg/overlay/cache.go

```diff
@@ -225,32 +225,32 @@ func (cache *Cache) FindStorageNodesWithPreferences(ctx context.Context, req Fin
 		reputableNodeCount = req.RequestedCount
 	}
 
-	auditCount := preferences.AuditCount
-	if auditCount < preferences.NewNodeAuditThreshold {
-		auditCount = preferences.NewNodeAuditThreshold
-	}
-
-	reputableNodes, err := cache.db.SelectStorageNodes(ctx, reputableNodeCount, &NodeCriteria{
+	newNodeCount := int(float64(reputableNodeCount) * preferences.NewNodePercentage)
```

Comment:
can you check if `preferences.NewNodePercentage > 0` otherwise set `newNodeCount = 0`?

Reply:
Makes sense. Done.